### PR TITLE
Refactor: Separate and Refactor Matching Logic into Utility Functions

### PR DESCRIPTION
**Pull request by saddamhr**
This PR refactors the keyboard event matching logic by moving it into utility functions to improve code organization and readability.

#### Key Changes:
- **Created `keyboard-utils.ts`**: 
  - Added `isMatchingKey` to handle key comparison (case-insensitive).
  - Added `isMatchingModifiers` to handle modifier key checks (meta, ctrl, shift, alt).

- **Refactored `useKeyboardShortcut` Hook**:
  - Replaced inline key and modifier logic with the new utility functions.
  - Maintained the same functionality, focusing on internal improvements.

#### Benefits:
- **Improved Code Organization**: The logic is now modular and easier to maintain.
- **Enhanced Testability**: Utility functions can be independently tested.

### CODE REVIEW

File: src/utils/keyboard-utils.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+const isMatchingModifiers = (
+  event: KeyboardEvent,
+  metaKey: boolean,
+  ctrlKey: boolean,
+  shiftKey: boolean,
+  altKey: boolean
+): boolean => {
+  return (
+    ((metaKey && event.metaKey) || (ctrlKey && event.ctrlKey)) &&
+    event.shiftKey === shiftKey &&
```
Comment (rifat-shaun) [inline src/utils/keyboard-utils.ts:31]: Looks better now. But can we do a small change here to make some consistency? for all condition checking either make it (eventKey === keyPassedAsProps) or vice versa

File: src/utils/keyboard-utils.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+ */
+const isMatchingModifiers = (
+  event: KeyboardEvent,
+  metaKey: boolean,
+  ctrlKey: boolean,
+  shiftKey: boolean,
+  altKey: boolean
+): boolean => {
+  return (
+    ((metaKey && event.metaKey) || (ctrlKey && event.ctrlKey)) &&
```
Comment (rifat-shaun) [inline src/utils/keyboard-utils.ts]: Looks better now. But can we do a small change here to make some consistency or better readability? for all condition checking either make it (eventKey === keyPassedAsProps) or vice versa

**Review by rifat-shaun — CHANGES_REQUESTED** [review]

File: src/utils/keyboard-utils.ts (hunk 2 above)
Comment (saddamhr) [reply src/utils/keyboard-utils.ts]: Thank you for your feedback. The change has been made, could you please check?

Comment (rifat-shaun) [reply src/utils/keyboard-utils.ts]: @saddamhr  Thanks for your contribution now we are good to go.

**Pull request merged**